src/components/collapsible-motion/collapsible-motion.js

```diff
@@ -17,144 +26,159 @@ const createClosingAnimation = height =>
     to { height: 0; overflow: hidden; }
   `;
 
-export class ToggleAnimation extends React.Component {
-  static displayName = 'ToggleAnimation';
-  static propTypes = {
-    isOpen: PropTypes.bool.isRequired,
-    toggle: PropTypes.func.isRequired,
-    children: PropTypes.func.isRequired,
-  };
-
-  static getDerivedStateFromProps(props, state) {
-    const containerStyles = props.isOpen
-      ? { height: 'auto' }
-      : { height: 0, overflow: 'hidden' };
-
-    let animation = null;
-    if (props.isOpen !== state.isOpen) {
-      if (props.isOpen) {
-        animation = createOpeningAnimation(state.fullHeight);
-      } else {
-        animation = createClosingAnimation(state.fullHeight);
-      }
-    }
-
-    return {
-      isOpen: props.isOpen,
-      containerStyles,
-      animation,
-    };
-  }
-
-  nodeRef = React.createRef();
-
-  state = {
-    isOpen: this.props.isOpen,
-    fullHeight: null,
-    animation: null,
-    containerStyles: this.props.isOpen
-      ? { height: 'auto' }
-      : { height: 0, overflow: 'hidden' },
-  };
-
-  componentDidMount() {
-    invariant(
-      this.nodeRef.current,
-      'You need to call `registerContentNode` in order to use this component'
-    );
-
-    // Sets the full height to the content's height.
-    // We can't set this when initializing the state as the element in nodeRef
-    // would not have mounted yet.
-    if (this.props.isOpen) {
-      this.setState({
-        fullHeight: this.nodeRef.current.clientHeight,
-      });
-    }
-  }
-
-  handleToggle = () => {
+const useToggleAnimation = (isOpen, toggle) => {
+  const nodeRef = React.useRef();
+  const prevIsOpen = usePrevious(isOpen);
+
+  React.useEffect(
+    () => {
+      invariant(
```

Comment:
We might want to wrap this into a dev mode condition. PropTypes are often compiled away. This not. So it throwing might break people's apps.

Reply:
this is the same behaviour as before -> the invariant was called in componentDidMount